examples/petsc/dmswarm.c

```diff
@@ -0,0 +1,990 @@
+  MPI_Comm  comm;
+  char      ceed_resource[PETSC_MAX_PATH_LEN] = "/cpu/self";
+  PetscBool test_mode = PETSC_FALSE, set_uniform_swarm = PETSC_FALSE, set_gauss_swarm = PETSC_FALSE, view_petsc_swarm = PETSC_FALSE,
+            view_ceed_swarm = PETSC_FALSE, use_polynomial_target = PETSC_FALSE;
+  PetscInt           dim = 3, num_comp = 1, num_points = 200, mesh_order = 1, solution_order = 3, q_extra = 3;
+  PetscScalar        tolerance = 1E-3;
+  DM                 dm_mesh, dm_swarm;
+  Vec                U_mesh;
+  DMSwarmCeedContext swarm_ceed_context;
+
+  PetscCall(PetscInitialize(&argc, &argv, NULL, help));
+  comm = PETSC_COMM_WORLD;
+
+  // Read command line options
+  PetscOptionsBegin(comm, NULL, "libCEED example using PETSc with DMSwarm", NULL);
+
+  PetscCall(PetscOptionsBool("-test", "Testing mode (do not print unless error is large)", NULL, test_mode, &test_mode, NULL));
+  PetscCall(PetscOptionsBool("-gauss_swarm", "Use gauss points for coordinates in swarm", NULL, set_gauss_swarm, &set_gauss_swarm, NULL));
+  PetscCall(PetscOptionsBool("-uniform_swarm", "Use uniform coordinates in swarm", NULL, set_uniform_swarm, &set_uniform_swarm, NULL));
+  PetscCall(
+      PetscOptionsBool("-u_petsc_swarm_view", "View XDMF of swarm values interpolated by PETSc", NULL, view_petsc_swarm, &view_petsc_swarm, NULL));
+  PetscCall(
+      PetscOptionsBool("-u_ceed_swarm_view", "View XDMF of swarm values interpolated by libCEED", NULL, view_ceed_swarm, &view_ceed_swarm, NULL));
+  PetscCall(PetscOptionsBool("-polynomial_target", "Use polynomial target function in field instead of tanh()", NULL, use_polynomial_target,
+                             &use_polynomial_target, NULL));
+  PetscCall(PetscOptionsInt("-solution_order", "Order of mesh solution space", NULL, solution_order, &solution_order, NULL));
+  PetscCall(PetscOptionsInt("-mesh_order", "Order of mesh coordinate space", NULL, mesh_order, &mesh_order, NULL));
+  PetscCall(PetscOptionsInt("-q_extra", "Number of extra quadrature points", NULL, q_extra, &q_extra, NULL));
+  PetscCall(PetscOptionsInt("-num_comp", "Number of components in solution", NULL, num_comp, &num_comp, NULL));
+  {
+    PetscInt dim = 3, num_cells[] = {1, 1, 1};
+
+    PetscCall(PetscOptionsInt("-dm_plex_dim", "Background mesh dimension", NULL, dim, &dim, NULL));
+    PetscCall(PetscOptionsIntArray("-dm_plex_box_faces", "Number of cells", NULL, num_cells, &dim, NULL));
+    PetscInt total_num_cells = num_cells[0] * num_cells[1] * num_cells[2];
+    PetscInt points_per_cell = 1;
+
+    for (PetscInt i = 0; i < dim; i++) points_per_cell *= solution_order + 2;
+    num_points = 0;
+    for (PetscInt i = 0; i < total_num_cells; i++) num_points += points_per_cell;
+  }
+  PetscCall(PetscOptionsInt("-points", "Number of swarm points", NULL, num_points, &num_points, NULL));
+  PetscCall(PetscOptionsScalar("-tolerance", "Absolute tolerance for swarm point values", NULL, tolerance, &tolerance, NULL));
```

Comment:
```suggestion
  PetscCall(PetscOptionsScalar("-tolerance", "Tolerance for swarm point values and projection relative L2 error", NULL, tolerance, &tolerance, NULL));
```
Should make it clear that we are using this differently for point values and projection error